### ISSUE TITLE
Something confusing to users

### ISSUE THREAD

**Issue by wwwwwwwwbh** [faneshion/MatchZoo#194]
Hi, contributors.
I am using your tool these days. It is really an excellent tool for beginners. 
However, I found things confused me while reading the source code.

First, the variable        **display_interval**      in config file is used as steps per epoch in function model.fit when training. 
Display interval sounds like a variable to print something which is not very important. 
I think it would be better to change a name.

Secondly, in the training step, the data generator choose samples from relation file randomly. I think the standard process should be implemented as: cover all dataset batch by batch, after applying shuffle if applicable. Random sampling may result in overfitting issue.

Thirdly, at the end of model drmm_tks, maybe same issue in other models 
out_ = Dense(2, activation='softmax')(mean)
The final layer output only have two values. How to achieve multiple classification by this?

Finally, in main.py, the function overlapping the embedding of the final word if vocab_size equals to the length of word embedding dict.
embed_dict[_PAD_] = np.zeros((share_input_conf['embed_size'], ), dtype=np.float32)

By the way, I have solve the second issue myself. if you need the code please contact me.

Rgds
Baohui

**Comment by bwanglzu** [comment]
hi @wwwwwwwwbh Thanks for your advice.
For your four quesionts & advices, I think @faneshion is the best one to answer.

We're currently working on MatchZoo 2.0, which will provide clear API on how to perform processing, training and evaluation!

Best!

**Comment by faneshion** [comment]
Thanks for your advices @wwwwwwwwbh . There are indeed ambiguities in some parameters. As @bwanglzu said, we are developing the MatchZoo 2.0 to keep it clear and effective. By the way, for your second question, in the training phase, we choose to randomly sample the instance. This is due to  the fact that in some tasks there are large number of training samples, and it is no need to go through all training samples. Both of these two training mechanism is useful, we will make it more flexible in the MatchZoo 2.0.

**Comment by wwwwwwwwbh** [comment]
Thank you very much for this clear and understandable answers！


Rgds

**Comment by xxllp** [comment]
我很好奇这个里面的模型训练出来的结果是怎么保存的？

**Comment by bwanglzu** [comment]
@xxllp 
Please refer to this [line](https://github.com/faneshion/MatchZoo/blob/47120ddd83d84623ed0f6eb49404854a579e6af8/matchzoo/main.py#L181), we use `keras.save_wegiths` to save the model weights into a HDF5 file.

In matchzoo 2.0, it's gonna be much easier to use our functional api:

```python
# save model
model.save(path)
# load model
from matchzoo import engine
model = engine.load_model(path)
```

**Comment by xxllp** [comment]
@bwanglzu ， I see ，and  how to get the result of the predict  text ?

**Comment by bwanglzu** [comment]
@xxllp I'm not familar with the previous version of matchzoo as well, but for the MatchZoo 2.0, checkout [this tutorial](http://nbviewer.jupyter.org/github/faneshion/MatchZoo/blob/2.0/notebooks/wikiqa/dssm.ipynb).

Matchzoo 2.0 is still under development, but you can checkout to 2.0 branch to take a look.

**Comment by xxllp** [comment]
thanks ,I will have a look
